### ISSUE TITLE
BalancedBatches for labeled data

### ISSUE THREAD

**Issue by Evizero** [JuliaML/MLDataPattern.jl#4]
Would be nice to have a new data iterator that samples the given data in such a way, that each iteration a batch is returned that contains an equal amount of observations from each class (no matter the class distribution)

```julia
X = rand(2, 6) # some features
y = [:a, :a, :a, :a, :b, :b]
for (xbatch, ybatch) in BalancedBatches((X, y), size = 2, count = 10)
    # ybatch is always either [:a, :b] or [:b, :a]
end
```

**Comment by Sheemon7** [comment]
Agreed. I believe that now when `RandomObs`/`RandomBatches` are available, it makes sense to have `BalancedBatches` in addition to `BalancedObs` as well.
